src/fluree/server/handler.clj

```diff
@@ -255,38 +256,74 @@
-                                                  {:status 400}))))
-                              (catch Exception _
-                                (throw (ex-info "Invalid Fluree-Policy-Values header: must be JSON."
-                                                {:status 400})))))]
-      (handler
-       (cond-> req
-         policy-identity (assoc :policy/identity policy-identity)
-         policy-class (assoc :policy/class policy-class)
-         policy (assoc :policy/policy policy)
-         policy-values (assoc :policy/values policy-values))))))
+  (fn [{:keys [headers credential/did] :as req}]
+    (let [{:keys [meta max-fuel identity policy-identity policy policy-class policy-values format output]}
+          (-> headers
+              (select-keys ["fluree-meta" "fluree-max-fuel" "fluree-identity" "fluree-policy-identity"
+                            "fluree-policy" "fluree-policy-class" "fluree-policy-values"
+                            "fluree-format" "fluree-output"])
+              (update-keys (fn [k] (keyword (subs k (count "fluree-"))))))
+
+          meta     (when meta
+                     (case meta
+                       "false" false
+                       "true"  true
+                       (throw (ex-info "Invalid Fluree-Meta header: must be boolean."
+                                       {:status 400}))))
+          max-fuel (when max-fuel
+                     (try (Integer/parseInt max-fuel)
+                          (catch Exception _
+                            (throw (ex-info "Invalid Fluree-Max-Fuel header: must be integer."
+                                            {:status 400})))))
+          ;; Accept header takes precedence over other ways of specifying query output
+          output   (cond (-> headers (get "accept") (= "application/sparql-results+json"))
+                         :sparql
+
+                         (= output "sparql") :sparql
+                         (= output "fql")    :fql
+                         :else               :fql)
+          ;; Content-Type header takes precedence over other ways of specifying query format
+          format        (cond (-> headers (get "content-type") (= "application/sparql-query"))
+                              :sparql
+
+                              (= format "sparql") :sparql
+                              (= output "fql")    :fql
+                              :else               :fql)
+          policy        (when policy
+                          (try (json/parse policy false)
+                               (catch Exception _
+                                 (throw (ex-info "Invalid Fluree-Policy header: must be JSON."
+                                                 {:status 400})))))
+          policy-values (when policy-values
+                          (try (let [pv (json/parse policy-values false)]
+                                 (if (sequential? pv)
```

Comment:
Does it make sense to validate this with malli since we already have a validator for that defined?

Reply:
I think the original idea was that the whole query will be run through the coercer/validator down the line, so we just do a very cheap check right here. If it's malformed we'll get a proper error message when the whole query is checked.